stock_picking_print_traceability/models/stock_picking.py

```diff
@@ -0,0 +1,31 @@
+from odoo import api, fields, models
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    print_date = fields.Datetime(
+        string="Print Date",
+        readonly=True,
+        help="Date on which stock picking is printed.",
+    )
+
+    @api.multi
+    def do_print_picking(self):
+        """Handles 'Print' button."""
+        res = super(Picking, self).do_print_picking()
+        self.set_printed(fields.Datetime.now())
+        return res
+
+    @api.multi
+    def set_printed(self, date):
+        self.ensure_one()
+        self.write({"printed": True, "print_date": date})  # `printed`: standard field
+
+    @api.multi
+    def write(self, vals):
+        res = super(Picking, self).write(vals)
+        if "printed" not in vals:
+            for picking in self:
+                picking.printed = False
+        return res

```

Comment:
It seems quite brutal to catch _every_ write but I understand it seems quite arbitrary to select only a few arbitrary fields to watch.

Question: can you have a write with an empty "vals" if the user clicks "edit" and "save" without modification?

The loop can be replaced by either 

```py
picking.write({"printed": True})
```

or, better, to avoid the recursion and sql queries
-> add to vals before super:

```py
def write(self, vals):
  if "printed" not in vals:
      vals["printed"] = False
  return super().write(vals)
```

Reply:
> can you have a write with an empty "vals" if the user clicks "edit" and "save" without modification?

No, there is no call to `write()` in that case.

Reply:
Thank you for the much better solution !